packages/libs/wdk-client/src/Core/routes.tsx

```diff
@@ -325,3 +304,40 @@ function parseSearchParamsFromQueryParams(restQueryParams: {
     ? Object.fromEntries(initialParamValuesEntries)
     : undefined;
 }
+
+/**
+ * Creates a route component for AnswerController with proper query string handling
+ * @param routeMatches - Optional. When provided, uses the specified recordClass and question
+ *                       instead of extracting them from route parameters. Useful for hardcoding
+ *                       specific routes (e.g., {recordClass: 'dataset', question: 'AllDatasets'}).
+ *                       When omitted, the recordClass and question are extracted from the route's
+ *                       :recordClass and :question path parameters.
+ * @returns A route component function
+ */
+export function makeAnswerControllerRouteComponent(routeMatches?: {
+  recordClass: string;
+  question: string;
+}) {
+  return (
+    props: RouteComponentProps<{ recordClass: string; question: string }>
+  ) => {
+    const {
+      filterTerm,
+      filterAttributes = [],
+      filterTables = [],
+    } = QueryString.parse(props.location.search.slice(1));
+    const parameters = parseSearchParamsFromQueryParams(
+      parseQueryString(props)
+    );
+    return (
+      <AnswerController
+        {...(routeMatches ?? props.match.params)}
+        parameters={parameters}
+        filterTerm={isArray(filterTerm) ? filterTerm[0] : filterTerm}
+        filterAttributes={castArray(filterAttributes)}
+        filterTables={castArray(filterTables)}
+        history={props.history}
```

Comment:
For me in my local dev server, it was necessary to pass all these props to `AnswerController` so that table search worked without a null `history` exception.